linfa-supervised/examples/main.rs

```diff
@@ -0,0 +1,40 @@
+use linfa_supervised::LinearRegression;
+use linfa_supervised::RidgeRegression;
+use ndarray::array;
+
+fn linear_regression() {
+    let mut linear_regression = LinearRegression::new(false);
+    let x = array![[1.0], [2.0], [3.0], [4.0]];
+    let y = array![1.0, 2.0, 3.0, 4.0];
+    linear_regression.fit(&x, &y);
+    let x_hat = array![[6.0], [7.0]];
+    println!("{:#?}", linear_regression.predict(&x_hat));
+
+    let mut linear_regression2 = LinearRegression::new(true);
```

Comment:
I would suggest splitting this example (and the one below) in two separate examples, with a descriptive name (e.g. with/without intercept).
It would be ideal to have them in separate files as well under the `examples` folder.